docs/x509.rst

```diff
@@ -735,6 +735,50 @@ X.509 Extensions
     extension is only relevant when the certificate is an authorized OCSP
     responder.
 
+.. class:: NameConstraints
+
+    .. versionadded:: 1.0
+
+    The name constraints extension, which must be used only in a CA
+    certificate, indicates a name space within which all subject names in
+    subsequent certificates in a certification path must be located.
+    Restrictions apply to the subject distinguished name and apply to
+    subject alternative names.  Restrictions apply only when the
+    specified name form is present.  If no name of the type is in the
+    certificate, the certificate is acceptable. For additional details on
+    the way this extension should be processed see :rfc:`5280`.
+
+    .. attribute:: permitted_subtrees
+
+        :type: list of :class:`GeneralSubtree` objects or None
+
+        The set of permitted name patterns. If a name matches this and an
+        element in ``excluded_subtrees`` it is invalid. ``permitted_subtrees``
+        and ``excluded_subtrees``. At least one of ``permitted_subtrees`` and
+        ``excluded_subtrees`` will be non-None.
+
+    .. attribute:: excluded_subtrees
+
+        :type: list of :class:`GeneralSubtree` objects or None
+
+        Any name matching a restriction in the ``excluded_subtrees`` field is
+        invalid regardless of information appearing in the
+        ``permitted_subtrees``. At least one of ``permitted_subtrees`` and
+        ``excluded_subtrees`` will be non-None.
+
+.. class:: GeneralSubtree
+
+    .. versionadded:: 1.0
+
+    A helper object used in :class:`NameConstraints`.
+
+    .. attribute:: base
+
+        :type: :class:`GeneralName`
+
+        The name used as the base name pattern for permitted or excluded
+        matching.
```

Comment:
What value does this object add above just putting a `GeneralName` in the lists on `NameConstraints`?